UndertaleModLib/Models/UndertaleEmbeddedTexture.cs

```diff
@@ -101,25 +103,82 @@ public override string ToString()
         return Name.Content + " (" + GetType().Name + ")";
     }
 
+    /// <inheritdoc/>
+    public void Dispose()
+    {
+        GC.SuppressFinalize(this);
+
+        TextureData.Dispose();
+        Name = null;
+    }
+
     /// <summary>
     /// Texture data in an <see cref="UndertaleEmbeddedTexture"/>.
     /// </summary>
-    public class TexData : UndertaleObject, INotifyPropertyChanged
+    public class TexData : UndertaleObject, INotifyPropertyChanged, IDisposable
     {
         private byte[] _textureBlob;
+        private int _width, _height;
+        private static MemoryStream sharedStream;
 
         /// <summary>
         /// The image data of the texture.
         /// </summary>
-        public byte[] TextureBlob { get => _textureBlob; set { _textureBlob = value; PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(nameof(TextureBlob))); } }
+        public byte[] TextureBlob
+        {
+            get => _textureBlob;
+            set
+            {
+                _textureBlob = value;
+                OnPropertyChanged();
+            }
+        }
+
+        /// <summary>
+        /// A texture width.
```

Comment:
```suggestion
        /// The width of the texture.
```

Reply:
Then, isn't _"The texture width"_ better?

Reply:
nah that sounds weirder imo